src/components/ContentSidebar/ActivityFeed/task/Task.js

```diff
@@ -115,35 +111,38 @@ class Task extends React.Component<Props> {
-                                case TASK_INCOMPLETE:
-                                    return (
-                                        <PendingAssignment
-                                            {...assigneeUser}
-                                            key={assigneeUser.id}
-                                            onTaskApproval={() =>
-                                                onTaskAssignmentUpdate(id, taskAssignmentId, TASK_APPROVED)
-                                            }
-                                            onTaskReject={() =>
-                                                onTaskAssignmentUpdate(id, taskAssignmentId, TASK_REJECTED)
-                                            }
-                                            shouldShowActions={
-                                                onTaskAssignmentUpdate &&
-                                                currentUser &&
-                                                assigneeUser.id === currentUser.id
-                                            }
-                                        />
-                                    );
-                                case TASK_COMPLETED:
-                                case TASK_APPROVED:
-                                    return <CompletedAssignment {...assigneeUser} key={assigneeUser.id} />;
-                                case TASK_REJECTED:
-                                    return <RejectedAssignment {...assigneeUser} key={assigneeUser.id} />;
-                                default:
-                                    return null;
-                            }
-                        })}
+                        {task_assignment_collection && task_assignment_collection.entries
+                            ? task_assignment_collection.entries.map(
+                                ({ id: assignmentId, assigned_to, resolution_state }) => {
+                                    switch (resolution_state) {
+                                        case TASK_COMPLETED:
+                                        case TASK_APPROVED:
+                                            return <CompletedAssignment {...assigned_to} key={assigned_to.id} />;
+                                        case TASK_REJECTED:
+                                            return <RejectedAssignment {...assigned_to} key={assigned_to.id} />;
+                                        case TASK_INCOMPLETE:
+                                        default:
+                                            return (
+                                                <PendingAssignment
+                                                    {...assigned_to}
+                                                    key={assigned_to.id}
+                                                    onTaskApproval={() =>
+                                                        onTaskAssignmentUpdate(id, assignmentId, TASK_APPROVED)
```

Comment:
Can you make sure that when you extract `onTaskAssigment` from props, you default to `noop`? Just to make sure this doesn't blow up

Reply:
👍